src/api/utils/tradeQuoter.ts

```diff
@@ -312,8 +320,14 @@ export class TradeQuoter {
       (feePercentage / 100)
     );
 
+
     const fromTokenAmount = quote.sellAmount;
 
+    let fromUnits;
+    if (fromTokenRequestAmount.eq(fromTokenAmount)) {
```

Comment:
You cannot use `fromTokenRequestAmount`. Compare `fromTokenAmount` to the max value implied by the position. If so, return the position's `units` which are then passed to the TradeModule (you are trying to trade the entire position but can't account for the precision)